app/controllers/greetings_controller.rb

```diff
@@ -0,0 +1,74 @@
+class GreetingsController < ApplicationController
+  before_action :set_greeting, only: %i[show edit update destroy]
+
+  # GET /greetings or /greetings.json
+  def index
+    @greetings = Greeting.all
+  end
+
+  # GET /greetings/1 or /greetings/1.json
+  def show; end
+
+  # GET /greetings/new
+  def new
+    @greeting = Greeting.new
+  end
+
+  # GET /greetings/1/edit
+  def edit; end
+
+  # POST /greetings or /greetings.json
+  def create
+    @greeting = Greeting.new(greeting_params)
+
+    respond_to do |format|
+      if @greeting.save
+        format.html { redirect_to greeting_url(@greeting), notice: 'Greeting was successfully created.' }
+        format.json { render :show, status: :created, location: @greeting }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @greeting.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /greetings/1 or /greetings/1.json
+  def update
+    respond_to do |format|
+      if @greeting.update(greeting_params)
+        format.html { redirect_to greeting_url(@greeting), notice: 'Greeting was successfully updated.' }
+        format.json { render :show, status: :ok, location: @greeting }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @greeting.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /greetings/1 or /greetings/1.json
+  def destroy
+    @greeting.destroy
+
+    respond_to do |format|
+      format.html { redirect_to greetings_url, notice: 'Greeting was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  def random
+    @greeting = Greeting.order('RANDOM()').first
+    render json: { greeting: @greeting.message }
+  end
```

Comment:
[OPTIONAL] Well done you did all changes given by the previous reviewer :point_up: but still there are some issues in your project if you run your project and go to the greeting page the greeting is not showing when you check your console it gives you an error `500 Server Error` and if you check your network the following error is coming so kindly fix this error as well :slightly_smiling_face: 

**_Network Error_**
![Screenshot from 2023-08-07 16-36-49](https://github.com/SabaAhmad404/Hello-rails-react/assets/45226848/7a326563-1074-4f08-a618-04a7918f62ff)

Reply:
hey @aamirkhan2478 wow i dont know why these greetings arent shown in your machine but my project working perfectly in my machine.